### PR TITLE
Fix option sizeof and write

### PR DESCRIPTION
**Pull request by Karang**

### CODE REVIEW

**Comment by rom1504** [comment]
Do you think it's possible to add a regression test for this there https://github.com/ProtoDef-io/ProtoDef/blob/master/test/conditional.json ?

**Comment by rom1504** [comment]
I think we should probably be iso to https://github.com/ProtoDef-io/node-protodef/blob/master/src/datatypes/conditional.js#L47 here. (Even if this behavior is a bit weird, relying on js implicit casting for null and undefined)

**Comment by Karang** [comment]
> Do you think it's possible to add a regression test for this there https://github.com/ProtoDef-io/ProtoDef/blob/master/test/conditional.json ?

Yes the case that was not tested was the "option not present" one. So basically an undefined value and an empty buffer.

**Comment by rom1504** [comment]
Could you add that ?

**Comment by rom1504** [comment]
Maybe, I'm not sure, try it

**Comment by Karang** [comment]
> Maybe, I'm not sure, try it

it works with null but we cannot test undefined

File: src/datatypes/compiler-conditional.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ module.exports = {
       return compiler.wrapCode(code, args)
     }],
     'option': ['parametrizable', (compiler, type) => {
-      let code = 'if (value !== null) {\n'
+      let code = 'if (value) {\n'
```
Comment (rom1504) [inline src/datatypes/compiler-conditional.js:50]: This should be != null, it's not equivalent

Comment (Karang) [reply src/datatypes/compiler-conditional.js]: done

**Pull request merged**